src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/LinearAVF.java

```diff
@@ -35,20 +38,10 @@
   private ReversedLinearValueFunction nbMinNightValueFunction;
 
   /**
-   * The 10 next arguments gives the apartment characteristic subjective value weight in the
-   * calculation of the Apartment total subjective value
+   * The next argument gives the apartment characteristic subjective value weight in the calculation
+   * of the Apartment total subjective value
    */
-  private double floorAreaWeight;
-
-  private double nbBedroomsWeight;
-  private double nbSleepingWeight;
-  private double nbBathroomsWeight;
-  private double terraceWeight;
-  private double floorAreaTerraceWeight;
-  private double wifiSubjectiveValueWeight;
-  private double pricePerNightSubjectiveValueWeight;
-  private double nbMinNightSubjectiveValueWeight;
-  private double teleSubjectiveValueWeight;
+  private HashMap<Criterion, Double> weight;
```

Comment:
Nous pourrions ici mettre une EnumMap plutôt qu'une Hashmap.
Exemple : EnumMap<Criterion, Double> weight;
this.weight = new EnumMap<> (Criterion.class);

Reply:
Effectivement, merci 🙃
Fix in d475cb0bc1a476a05cfd0eb89a9d9013da5b7cda